solidity/test/Polymer7683.t.sol

```diff
@@ -0,0 +1,392 @@
+            emitter,
+            topics,
+            data
+        );
+    
+        return "dummy_proof";
+    }
+
+    function test_setDestinationContract() public {
+        vm.prank(owner);
+    
+        polymer7683.setDestinationContract(destChainId, destContract);
+        address stored = polymer7683.destinationContracts(destChainId);
+        assertEq(stored, destContract, "Destination contract address mismatch");
+    }
+
+    function test_handleSettlementWithProof_success() public {
+        // Register destination contract
+        vm.prank(owner);
+        polymer7683.setDestinationContract(destChainId, destContract);
+    
+        console2.log("\nTest setup:");
+        console2.log("Local Chain ID:", localChainId);
+        console2.log("Destination Chain ID:", destChainId);
+        console2.log("Destination Contract:", destContract);
+    
+        // Create sample order ID and filler data
+        bytes32 orderId = bytes32("orderId1");
+        bytes memory fillerData = abi.encode(bytes32("filler1"));
+    
+        console2.log("\nOrder ID:");
+        console2.logBytes32(orderId);
+    
+        // Create and submit settlement proof
+        bytes memory proof = _createSettlementProof(destChainId, destContract, orderId, fillerData);
+    
+        console2.log("\nSettlement proof details:");
+        console2.log("Proof length:", proof.length);
+    
+        // Get proof validation details
+        (
+            uint32 provenChainId,
+            address emitter,
+            bytes memory topics,
+            bytes memory data
+        ) = prover.validateEvent(proof);
+        console2.log("Proven Chain ID:", provenChainId);
+        console2.log("Emitter:", emitter);
+        console2.log("Topics length:", topics.length);
+        console2.log("Data length:", data.length);
+
+        console2.log("\nDecoding event data...");
+        (
+            bytes32 decodedOrderId,
+            bytes memory originData,
+            bytes memory decodedFillerData
+        ) = abi.decode(data, (bytes32, bytes, bytes));
+        console2.log("Decoded Order ID:");
+        console2.logBytes32(decodedOrderId);
+        console2.log("Origin Data length:", originData.length);
+        console2.log("Filler Data length:", decodedFillerData.length);
+
+        console2.log("\nSubmitting settlement proof...");
+
+        try polymer7683.handleSettlementWithProof(proof) {
+            console2.log("\nCall succeeded");
+        } catch Error(string memory reason) {
+            console2.log("\nReverted with reason:", reason);
+        } catch (bytes memory errData) {
+            console2.log("\nReverted with raw error data length:", errData.length);
+            if (errData.length >= 4) {
+                bytes4 selector = bytes4(errData);
+                console2.log("Error selector:");
+                console2.logBytes4(selector);
+            }
+        }
+    }
+
+    function test_handleSettlementWithProof_preventReplay() public {
```

Comment:
This one is testing the `handleRefundWithProof` instead of the `handleSettlementWithProof`